solutions/eocode/solution.py

```diff
@@ -0,0 +1,37 @@
+"""Challenge 9"""
+from typing import List
+
+
+class Solution:
+    """Class of challenge solution."""
+
+    def duplicate_zeros(self, arr: List[int]):
+        """Function that duplicates 0s without altering the memory used."""
+        zeros = 0
+        a_len = len(arr)
+        secureZone = -1
+        for i, v in enumerate(arr):
+            if v == 0:
+                secureZone = i if secureZone == -1 else secureZone
+                if i + zeros + 1 < a_len:
+                    zeros += 1
+                else:
+                    break
+        b = a_len - 1
+        l = b - zeros
```

Comment:
Please use descriptive variable names.

Google's Python style guide:
http://google.github.io/styleguide/pyguide.html#316-naming

Reply:
Sure, I correct the variable names